src/subquery.js

```diff
@@ -0,0 +1,43 @@
+import _ from 'lodash/fp'
+import { getTypePropOrError } from './types'
+
+// A subquery (in contexture-client) is about taking the output of one search and makng it the input for another search.
+// This is an in memory, cross-database, "select in" join on sources that don't need to be relational.
```

Comment:
yes! yes yes yes yes, meaningful comments, making complexity easier to handle <3